### PR TITLE
Add clang-tidy checks to the Build-Ascend CI job and fix errors

### PR DESCRIPTION
**Pull request by jfxu-st**
## Motivation and Context

For Ascend related code, there are no error detection and style checks based on static analysis.

## Description

- Fix a check format bug caused by a missing comma in .clang-tidy
- Remove redundant debug options in Ascend's CMakeLists.txt
- Remove soft links in Ascend test and update CMakeLists.txt accordingly 
- Fix errors detected by clang-tidy
- Add clang-tidy checks to the Build-Ascend CI job
- Remove useless Element related code in utils.cpp
- Refactor setAttr templates in acloprunner.hpp to improve readability 

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/common/utils.cpp [hunk 1]
```diff
@@ -241,13 +242,13 @@ bool isContiguous(diopiConstTensorHandle_t tensor, diopiMemoryFormat_t format) {
 
 class Element {
```
Comment (hellozmz) [inline impl/ascend/common/utils.cpp:243]: element应该没有使用到，直接删除掉吧。

Comment (jfxu-st) [reply impl/ascend/common/utils.cpp:243]: > element应该没有使用到，直接删除掉吧。

`Element` 被 `bool compare(Element& a, Element& b)` 用到了，把这俩一起删了？

Comment (hellozmz) [reply impl/ascend/common/utils.cpp:243]: 是的，都可以删除掉。应该是个临时代码。

Comment (jfxu-st) [reply impl/ascend/common/utils.cpp]: OK，已删除。

**Review by hellozmz — APPROVED** [review]

File: impl/ascend/common/acloprunner.hpp [hunk 2]
```diff
@@ -542,31 +542,39 @@ class AclOpRunner {
...
-        }
-        if constexpr (std::is_same<T, uint8_t>::value || std::is_same<T, bool>::value) {
-            CALL_ACLRT(aclopSetAttrBool(attr_, attrName.data(), value));
-            return *this;
-        }
-        if constexpr (std::is_same<T, std::string>::value) {
-            CALL_ACLRT(aclopSetAttrString(attr_, attrName.data(), value.data()));
-            return *this;
-        }
+    std::enable_if_t<std::is_same<T, int64_t>::value || std::is_same<T, int>::value, AclOpRunner&> setAttr(const std::string& attrName, const T& value) {
```
Comment (yangbofun) [inline impl/ascend/common/acloprunner.hpp:553]: 这个写法，不够好，std::enable_if_t可以提到template<T>里面

**Review by yangbofun — APPROVED** [review]

**Pull request merged**